File: fixtures/scale/dart/large/lib/ciphers/playfair/playfair_key.dart
class PlayfairKey {
  final String keyword;

  PlayfairKey(this.keyword);

  String square() {
    final sb = StringBuffer();
    final source = '${keyword.toUpperCase().replaceAll('J', 'I')}ABCDEFGHIKLMNOPQRSTUVWXYZ';
    for (final c in source.split('')) {
      if (c != 'J' && !sb.toString().contains(c)) sb.write(c);
    }
    return sb.toString();
  }

  factory PlayfairKey.defaultKey() => PlayfairKey('MONARCHY');
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/playfair/PlayfairKey.kt
package cryptobench.ciphers.playfair

data class PlayfairKey(val keyword: String) {

    fun square(): String {
        val sb = StringBuilder()
        for (c in keyword.uppercase().replace("J", "I") + "ABCDEFGHIKLMNOPQRSTUVWXYZ") {
            if (c in 'A'..'Z' && c != 'J' && !sb.contains(c)) sb.append(c)
        }
        return sb.toString()
    }

    companion object {
        fun default(): PlayfairKey = PlayfairKey("MONARCHY")
    }
}
